tq/transfer_queue.go

```diff
@@ -87,70 +87,101 @@ func (r *retryCounter) CanRetry(oid string) (int, bool) {
 	return count, count < r.MaxRetries
 }
 
+type Batch []Transferable
+
+func (b Batch) IsFull(max int) bool { return len(b) >= max }
```

Comment:
Yeah, I was a little hesitant about adding this function. I think it provides minimal value for the reason that you stated, and it expands the `Batch` interface. I've removed it in ced37c6c.